spec/lang/step/expressions.md

```diff
@@ -35,19 +35,19 @@ impl<M: Memory> Machine<M> {
             Constant::Bool(b) => Value::Bool(b),
             Constant::GlobalPointer(relocation) => {
                 let ptr = self.global_ptrs[relocation.name].wrapping_offset::<M::T>(relocation.offset.bytes());
-                Value::Ptr(ptr)
+                Value::Ptr(ptr.widen(None))
             },
             Constant::FnPointer(fn_name) => {
-                Value::Ptr(Pointer {
+                Value::Ptr((ThinPointer {
                     addr: self.fn_addrs[fn_name],
                     provenance: None,
-                })
+                }).widen(None))
```

Comment:
There seem to be unnecessary parentheses here? (And in a bunch of other places as well.)

Reply:
Oh yeah I thought they were necessary and usually rely on rust fmt to remove them if not. Guess I will actually learn rust syntax with this project xD